### PR TITLE
Adapted to VCF v4.2 specs ... 

### PR DESCRIPTION
**Pull request by temehi**
... 8 mandatory columns instead of 9

This fixes issue #2075  
Part of the test was removed because it was testing a random behavior.

### CODE REVIEW

File: tests/vcf_io/test_vcf_io.h [hunk 1]
```diff
@@ -161,12 +161,6 @@ SEQAN_DEFINE_TEST(test_vcf_io_read_vcf_record)
     SEQAN_ASSERT_EQ(records[2].format, "GT:GQ:DP:HQ");
     SEQAN_ASSERT_EQ(length(records[2].genotypeInfos), 3u);
 
-    for (unsigned i = 0; i < 25; ++i)
-    {
-        SEQAN_TEST_EXCEPTION(seqan::ParseError,
-                             seqan::readRecord(record, vcfIOContext, iter, seqan::Vcf()));
-        seqan::skipLine(iter);
-    }
```
Comment (rrahn) [inline tests/vcf_io/test_vcf_io.h]: Why is was this removed?

**Comment by rrahn** [comment]
@temehi ping

File: tests/vcf_io/test_vcf_io.h (hunk 1 above)
Comment (temehi) [reply tests/vcf_io/test_vcf_io.h]: I put it back. The problem was there were records in the test file which were invalid for vcf format v4.1 but now they are valid per the new (v4.2) specification. I removed those records from the test file.

File: tests/vcf_io/test_vcf_io.h [hunk 2]
```diff
@@ -161,11 +161,35 @@ SEQAN_DEFINE_TEST(test_vcf_io_read_vcf_record)
...
+    SEQAN_ASSERT_EQ(records[4].id, "rs6040355");
+    SEQAN_ASSERT_EQ(records[4].ref, "A");
+    SEQAN_ASSERT_EQ(records[4].alt, "G,T");
+    SEQAN_ASSERT_EQ(records[4].qual, 67);
+    SEQAN_ASSERT_EQ(records[4].filter, "PASS");
+    SEQAN_ASSERT_EQ(records[4].info, "NS=2;DP=10;AF=0.333,0.667;AA=T;DB");
+    SEQAN_ASSERT_EQ(records[4].format, ""); // empty formats are accepted since v4.2
+    SEQAN_ASSERT_EQ(length(records[4].genotypeInfos), 3u);
+
+    // the next 18 recodrs are invalid and readRecord should throw ParseError
```
Comment (rrahn) [inline tests/vcf_io/test_vcf_io.h]: typo: `records`

**Comment by h-2** [comment]
@rrahn you approved this PR, but didn't merge. Are there any TODOs left?

**Comment by rrahn** [comment]
one step after another.

**Pull request merged**